internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -191,6 +210,44 @@ func (r *BMACReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (c
 		return result.Result()
 	}
 
+	// Before reconciling agent spec, first check for the day2 flow
+	// if spoke cluster requirements are fulfilled.
+	fulfilled, clusterDeployment, err := r.spokeClusterRequirementsFulfilled(ctx, log, agent)
```

Comment:
all this logic should be inside ensureMCSCert and then you need to retrun result and validate it is dirty or not 
same as 
https://github.com/openshift/assisted-service/pull/1661/files#diff-86415614c6ed40c56c5ca2b5f751a65746ef5c83b61e6f021a4f5d817e4425baL181

Reply:
Done. Please check if this is what be expected.